tests/simulation_framework/test_entities_v2.py

```diff
@@ -50,219 +133,270 @@ def my_cylinder2():
-        assert "Input should be a valid dictionary or instance of Cylinder" in str(e)
-        assert "Input should be a valid string" in str(e)
-
+        assert (
+            f"Type(<class 'int'>) of input to `entities` (1) is not valid. Expected str or entity instance."
+            in str(e)
+        )
+    # 3. test empty list
     try:
-        expanded_entities = ZoneList(entities=[]).entities.get_expanded_entities()
+        expanded_entities = TempFluidDynamics(entities=[]).entities._get_expanded_entities()
     except ValueError as e:
         assert "Invalid input type to `entities`, list is empty." in str(e)
 
+    # 4. test None
     try:
-        expanded_entities = ZoneList(entities=None).entities.get_expanded_entities()
+        expanded_entities = TempFluidDynamics(entities=None).entities._get_expanded_entities()
     except pd.ValidationError as e:
-        assert "Value error, Invalid input type to `entities`: <class 'NoneType'>" in str(e)
+        assert (
+            "Type(<class 'NoneType'>) of input to `entities` (None) is not valid. Expected str or entity instance."
+            in str(e)
+        )
 
+    # 5. test non-existing entity
     try:
-        expanded_entities = ZoneList(
+        expanded_entities = TempFluidDynamics(
             entities=["Non_existing_volume"]
-        ).entities.get_expanded_entities()
+        ).entities._get_expanded_entities(EntityRegistry())
     except ValueError as e:
         assert (
             "Failed to find any matching entity with ['Non_existing_volume']. Please check the input to entities."
             in str(e)
         )
+    try:
+        my_volume_mesh1["asdf"]
+    except ValueError as e:
+        assert "Failed to find any matching entity with asdf. Please check your input." in str(e)
+
+
+def test_duplicate_entities(my_volume_mesh1):
+    user_override_cylinder = Cylinder(
+        name="zone_1",
+        height=12,
+        axis=(1, 0, 0),
+        inner_radius=1,
+        outer_radius=2,
+        center=(1, 2, 3),
+    )
+
+    expanded_entities = TempFluidDynamics(
+        entities=[
+            my_volume_mesh1["zone*"],
+            user_override_cylinder,
+            user_override_cylinder,
+            user_override_cylinder,
+            my_volume_mesh1["*"],
+        ]
+    ).entities._get_expanded_entities()
+
+    assert len(expanded_entities) == 3
+    assert user_override_cylinder in expanded_entities
+
+
+def test_entire_worklfow(my_cylinder1, my_volume_mesh1):
+
+    my_param = TempSimulationParam(
+        far_field_type="auto",
+        models=[
+            TempFluidDynamics(
+                entities=[
+                    my_cylinder1,
+                    my_cylinder1,
+                    my_cylinder1,
+                    my_volume_mesh1["*"],
+                    my_volume_mesh1["*zone*"],
+                ]
+            ),
+            TempWallBC(surfaces=[my_volume_mesh1["*"], "*", "farfield"]),
```

Comment:
This "*" actually raises concern that the user might use "*" thinking it is going to match everything when in reality it is only going to match surfaces in the Param's registry that contains implicitly defined surfaces.